unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Video/DCLVideoTexture.cs

```diff
@@ -187,9 +198,23 @@ private IEnumerator VideoTextureUpdate()
                     texturePlayer.UpdateWebVideoTexture();
                 }
 
+                if (texturePlayer.playing && IsTimeToReportVideoProgress())
```

Comment:
I think that this component logic is already getting pretty convoluted. We can't extract all the logic related to reporting the video progress to another class? (I don't mind if we have to create another coroutine for handling this).

Reply:
What about single level of abstraction? I prefer having less coroutines.
```c#
private IEnumerator OnUpdate()
{
    while (true)
    {
        UpdateDirtyState();
        UpdateVideoTexture();
        UpdateProgressReport();
        yield return null;
    }
}
```
Moving the report logic to another class could be another refactor or just leave it as technical debt

Reply:
Sounds good!